content/en/storage-providers/advanced-configurations/workers.md

```diff
@@ -0,0 +1,685 @@
+{{< /alert >}}
+
+### Resource windows
+
+The scheduler uses the concept of resource windows to prevent resource starvation of tasks requiring larger amounts of resources by tasks with smaller resource requirements.
+
+A resource window is simply a bucket of sealing tasks that can be run by a given worker in parallel based on the resources the worker has available when no tasks are running.
+
+In the scheduler, each worker has:
+- Scheduling windows - Two resource windows used to assign tasks to execute from the global queue
+- Preparing window - One resource window in which tasks are prepared to execute (for example, sector data is fetched if needed)
+- Executing window - One resource window for currently executing tasks
+
+When tasks arrive in the global scheduling queue, the scheduler will look for empty scheduling windows, and based on a number of factors, like whether the worker already has direct access to sector data, task types supported by the worker, whether the worker has disk space for sector data, task priority - tasks may be assigned to the scheduling window.
+
+After a scheduling window is filled with a number of tasks, it's sent to the worker for processing. The worker will pull tasks out of the scheduling window and start preparing them in the preparing window. After the preparing step is done, the task will be executed in the executing window.
+
+After the worker has fully processed a scheduling window, it's sent back to the global scheduler to get more sealing tasks.
+
+### Task priorities
+
+When the scheduler decides which tasks to run, it takes into account the priority of running a specific task.
+
+There are two priority tiers - high priority, for tasks that are cheap to execute but block other actions, and normal priority for all other tasks. Default priorities are defined in the table below.
+
+| Task Type           | Priority |
+|---------------------|----------|
+| RegenSectorKey      | 10       |
+| AddPiece            | 9        |
+| ReplicaUpdate       | 8        |
+| ProveReplicaUpdate2 | 7        |
+| ProveReplicaUpdate1 | 6        |
+| PreCommit1          | 5        |
+| PreCommit2          | 4        |
+| Commit2             | 3        |
+| Commit1             | 2        |
+| Unseal              | 1        |
+| Fetch               | -1       |
+| Finalize            | -2       |
+
+- Lower number means higher priority.
+- Negative number means the highest priority.
+
+When comparing task priority:
+- High priority tasks are considered first
+- Sectors with deals are considered second (more deals add more priority)
+- If the above is equal, tasks are selected based on priorities in the table
+- If the above is equal, sectors with lower sector numbers are selected (this can optimize gas usage slightly when submitting messages to the chain)
```

Comment:
```suggestion
When comparing task priority:

- High priority tasks are considered first.
- Sectors with deals are considered second. Additional deals increase the priority.
- If the above is equal, tasks are selected based on priorities in the table.
- If the above is equal, sectors with lower sector numbers are selected (this can optimize gas usage slightly when submitting messages to the chain).
```